packages/main/src/plugin/proxy.ts

```diff
@@ -178,15 +179,16 @@ export class Proxy {
     return this.proxyState !== ProxyState.PROXY_DISABLED && this.proxySettings !== undefined;
   }
 
-  setState(state: ProxyState): void {
+  async setState(state: ProxyState): Promise<void> {
     this.proxyState = state;
 
     // update configuration
     const proxyConfiguration = this.configurationRegistry.getConfiguration('proxy');
-    proxyConfiguration.update('enabled', state).catch((err: unknown) => {
-      console.error('Error updating proxy state', err);
-    });
+    await proxyConfiguration.update('enabled', state);
 
+    if (state === ProxyState.PROXY_SYSTEM) {
+      await this.setProxy({} as ProxySettings);
```

Comment:
I think setProxy should accept undefined as a parameter and here we call setProxy(undefined)